scalameta/semantic/shared/src/main/scala/scala/meta/internal/semantic/vfs/Paths.scala

```diff
@@ -32,11 +30,6 @@ object Paths {
 
   def scalaToSemanticdb(path: RelativePath): RelativePath = {
     require(isScala(path))
-    RelativePath(
-      semanticDbPrefix +
-        path.toString
-          .stripPrefix(scalaPrefix)
-          .stripSuffix(scalaSuffix) +
-        semanticDbSuffix)
+    semanticDbPrefix.resolve(path.resolveSibling(_.stripSuffix(scalaSuffix) + semanticDbSuffix))
```

Comment:
On a number of occasions, we have seen that moving from `String` to `*Path` brings correctness benefits. In general, I'm in favor of refactorings like that, as long as they are reasonably concise. I had problems with the first take on this, but I do like the new version. Thank you!